tests/test_e2e_40_sdntrace.py

```diff
@@ -942,3 +939,123 @@ def test_080_validate_attribute_on_payload(self):
         api_url = KYTOS_API + '/amlight/sdntrace_cp/v1/traces'
         response = requests.put(api_url, json=payload)
         assert response.status_code == 200
+
+    def test_085_test_evcs_terminating_on_nnis(cls):
+        "Test EVCs terminating on NNIs"
+
+        cid1 = cls.create_evc(999, "00:00:00:00:00:00:00:02:1", "00:00:00:00:00:00:00:04:1")
+        
+        payload = {
+                "name": "pw_s3",
+                "dynamic_backup_path": True,
+                "uni_a": {
+                    "interface_id": "00:00:00:00:00:00:00:03:2"
+                },
+                "uni_z": {
+                    "interface_id": "00:00:00:00:00:00:00:03:3"
+                }
+            }
+        api_url = KYTOS_API + '/kytos/mef_eline/v2/evc/'
+        response = requests.post(api_url, json=payload)
+        assert response.status_code == 201, response.text
+        cid2 = response.json()['circuit_id']
+        
+        payload = [
+                    {
+                        "trace": {
+                            "switch": {
+                                "dpid": "00:00:00:00:00:00:00:03",
+                                "in_port": 2
+                            }
+                        }
+                    }, {
+                        "trace": {
+                            "switch": {
+                                "dpid": "00:00:00:00:00:00:00:02",
+                                "in_port": 1
+                            },
+                            "eth": {"dl_vlan": 999}
+                        }
+                    }               
+                ]
+
+        api_url = KYTOS_API + '/amlight/sdntrace_cp/v1/traces'
+        response = requests.put(api_url, json=payload)
+        assert response.status_code == 200, response.text
+        results = response.json()['result']
+        data = results[0][0]
+        assert data['type'] == 'last'
+        assert data['dpid'] == '00:00:00:00:00:00:00:03'
+        assert data['port'] == 2
+        assert data['out']['port'] == 3
+
+        data = results[1]
+        assert data[0]['dpid'] == '00:00:00:00:00:00:00:02'
+        assert data[0]['port'] == 1
+        assert data[0]['type'] == 'starting'
+        assert data[0]['vlan'] == 999
+        assert data[-1]['dpid'] == '00:00:00:00:00:00:00:04'
+        assert data[-1]['out']['port'] == 1
+        assert data[-1]['type'] == 'last'
+        assert data[-1]['out']['vlan'] == 999
+
+        # Delete the created circuits 
+        api_url = KYTOS_API + f'/kytos/mef_eline/v2/evc/{cid1}' 
+        response = requests.delete(api_url)
+        assert response.status_code == 200, response.text
+
+        api_url = KYTOS_API + f'/kytos/mef_eline/v2/evc/{cid2}' 
+        response = requests.delete(api_url)
+        assert response.status_code == 200, response.text
+
+    def test_090_test_flows_with_instruction(cls):
```

Comment:
The logic of the test case looks good, but let's not forget to remove the flows since this test suite is still not cleaning up the DB after every test case (since it reuses EVCs).

Reply:
Now that you've included `self.net.start_controller(clean_config=True, enable_all=True)` on setup method, this is no longer a concern. Feel free to close this thread.